packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -33,6 +34,16 @@ - (void)onDisplayLink:(CADisplayLink *)link {
 }
 @end
 
+@interface AVPlayerFactory : NSObject <AVPlayerFactoryProtocol>
```

Comment:
Using an Apple prefix, like `AV`, for our own class, is a no-go in Objective-C, where prefixes serve as a sort of namespace; see https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/Conventions/Conventions.html
If Apple has (or later added) a class with this name, behavior at runtime would be undefined, and almost certainly crash.

This should be called `FLTDefaultAVPlayerFactory`, or something along those lines that begins with `FLT`, because every class we define in this plugin should begin with our own specific prefix. Or in fact, `FVP` per https://github.com/flutter/flutter/issues/102601; we can start shifting incrementally even though the main class is still FLT.